.github/workflows/ci.yaml

```diff
@@ -53,9 +53,11 @@ jobs:
           # from non default locations first. Installing the PyTorch CPU
           # wheels saves multiple minutes and a lot of bandwidth on runner setup.
           pip install --no-compile -r pytorch-cpu-requirements.txt
+          pip install --no-compile -r tuner-requirements.txt
```

Comment:
I think we don't (currently?) want the tuner to spread its dependencies to sharktank, so this probably should be kept separate. If the tuner becomes an integral part of sharktank then the deps should move into the base `requirements.txt` IMO.

Because of that, I suggest creating a separate job in this workflow file or creating a separate workflow file entirely, rather than add new dependencies here and try to run the tests in the same step sequence.

The main functional difference between the two options in this case is that separate workflow files can have different triggers, but you probably just want what is in this file:
```yml
on:
  workflow_dispatch:
  pull_request:
  push:
    branches:
      - main
```

## Option 1: stay in this file

```diff
jobs:
  test:
    name: "Unit Tests and Type Checking"
    strategy:
      matrix:
    ...

+  test_tuner:
+    name: "Unit Tests and Type Checking for Tuner"
+    ...  (copy/fork other steps as needed, just doing what is needed to run the tuner tests/lint)
```

## Option 2: create a new file

Like how https://github.com/nod-ai/sharktank/blob/main/.github/workflows/ci_linux_x64-libshortfin.yml is its own file, create a new `ci-tuner.yml`:

```yml
name: CI - Tuner
on:
  workflow_dispatch:
  pull_request:
  push:
    branches:
      - main
concurrency:
  # A PR number if a pull request and otherwise the commit hash. This cancels
  # queued and in-progress runs for the same PR (presubmit) or commit
  # (postsubmit). The workflow name is prepended to avoid conflicts between
  # different workflows.
  group: ${{ github.workflow }}-${{ github.event.number || github.sha }}
  cancel-in-progress: true
jobs:
  test:
    name: "Unit Tests and Type Checking"
    ...
```

Reply:
thanks, took the option 2